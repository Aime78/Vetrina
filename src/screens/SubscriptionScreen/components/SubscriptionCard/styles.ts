import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    marginBottom: 20,
    backgroundColor: 'white',
    borderRadius: 10,
    width: 370,
    height: 140,
    padding: 20,
    justifyContent: 'space-between',
  },
  shadow: {
    shadowColor: '#171717',
    shadowOffset: {width: 1, height: 2},
    shadowOpacity: 0.1,
    shadowRadius: 3,
  },
  containerHeader: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  containerBody: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  containerChange: {
    width: 118,
    height: 46,
    borderRadius: 5,
    backgroundColor: '#21B8F9',
    justifyContent: 'center',
    alignItems: 'center',
  },
  textFree: {
    fontWeight: '600',
    fontSize: 20,
    color: '#103B66',
  },
  textMax: {
    fontWeight: '600',
    fontSize: 15,
    color: '#103B66',
  },
  textCurrency: {
    fontWeight: '600',
    fontSize: 17,
    color: '#103B66',
  },
  textNumber: {
    fontWeight: '600',
    fontSize: 28,
    color: '#21B8F9',
  },
  textChange: {
    fontWeight: '600',
    fontSize: 16,
    color: '#FFFFFF',
  },
  textMessi: {
    fontWeight: '600',
    fontSize: 12,
    color: '#00C48C',
    marginTop: 10,
  },
});

export default styles;
