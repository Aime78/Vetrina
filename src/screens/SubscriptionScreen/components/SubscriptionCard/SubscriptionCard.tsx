import {View, Text} from 'react-native';
import styles from './styles';

type SubscriptionCardProps = {
  name: string;
  price: string;
  messi: string;
};
const SubscriptionCard = ({name, price, messi}: SubscriptionCardProps) => {
  return (
    <View style={[styles.container, styles.shadow]}>
      <View style={styles.containerHeader}>
        <Text style={styles.textFree}>{name}</Text>
        <Text style={styles.textMax}>Max 150 prodotti</Text>
      </View>
      <Text style={styles.textMessi}>{messi}</Text>
      <View style={styles.containerBody}>
        <Text style={styles.textCurrency}>
          â‚¬ <Text style={styles.textNumber}>{price}</Text>/year
        </Text>
        <View style={styles.containerChange}>
          <Text style={styles.textChange}>Change plan</Text>
        </View>
      </View>
    </View>
  );
};

export default SubscriptionCard;
