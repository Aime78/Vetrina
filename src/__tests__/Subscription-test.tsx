import React from 'react';
import renderer from 'react-test-renderer';
import methods from 'screens/PaymentScreen/cardData';
import SubscriptionCard from 'screens/SubscriptionScreen/components/SubscriptionCard/SubscriptionCard';
import plans from 'screens/SubscriptionScreen/subcriptionData';

it('renders the all subscription plans of the subscription screen appropriately', () => {
  const tree = renderer
    .create(
      <SubscriptionCard
        name={plans[0].name}
        price={plans[0].number}
        messi={plans[0].messi}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
