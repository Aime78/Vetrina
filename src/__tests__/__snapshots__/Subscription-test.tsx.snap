// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the all subscription plans of the subscription screen appropriately 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": 140,
        "justifyContent": "space-between",
        "marginBottom": 20,
        "padding": 20,
        "width": 370,
      },
      {
        "shadowColor": "#171717",
        "shadowOffset": {
          "height": 2,
          "width": 1,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        {
          "color": "#103B66",
          "fontSize": 20,
          "fontWeight": "600",
        }
      }
    >
      Free
    </Text>
    <Text
      style={
        {
          "color": "#103B66",
          "fontSize": 15,
          "fontWeight": "600",
        }
      }
    >
      Max 150 prodotti
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#00C48C",
        "fontSize": 12,
        "fontWeight": "600",
        "marginTop": 10,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        {
          "color": "#103B66",
          "fontSize": 17,
          "fontWeight": "600",
        }
      }
    >
      â‚¬ 
      <Text
        style={
          {
            "color": "#21B8F9",
            "fontSize": 28,
            "fontWeight": "600",
          }
        }
      >
        0.00
      </Text>
      /year
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#21B8F9",
          "borderRadius": 5,
          "height": 46,
          "justifyContent": "center",
          "width": 118,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Change plan
      </Text>
    </View>
  </View>
</View>
`;
